util-images/network/netperfbenchmark/pkg/worker/worker.go

```diff
@@ -0,0 +1,458 @@
+	//// siege http://localhost:5301/test -d1 -r1 -c1 -t10S
+	//c concurrent r repetitions t time d delay in sec between 1 and d
+	klog.Info("In StartHTTPClient")
+	ts, dur, destIP, _ := parseURLParam(req)
+	if dur == "" || destIP == "" {
+		createResp(startWrkResponse{error: "missing/invalid required parameters"}, &res)
+		return
+	}
+	go schedule(protocolHTTP, ts, dur, "siege",
+		[]string{"http://" + destIP + ":" + httpPort + "/test",
+			"-d1", "-t" + dur + "S", "-c1"})
+	createResp(startWrkResponse{}, &res)
+}
+
+//Handler handles http requests for http measurements
+func Handler(res http.ResponseWriter, req *http.Request) {
+	fmt.Fprintf(res, "hi\n")
+}
+
+func parseURLParam(req *http.Request) (int64, string, string, string) {
+	values := req.URL.Query()
+	ts := values.Get("timestamp")
+	dur := values.Get("duration")
+	dstIP := values.Get("destIP")
+	numcl := values.Get("numCls")
+	var tsint int64
+	var err error
+	if ts != "" {
+		tsint, err = strconv.ParseInt(ts, 10, 64)
+		if err != nil {
+			klog.Info("Invalid timestamp:", ts, " ", err)
+		}
+	}
+	return tsint, dur, dstIP, numcl
+
+}
+
+func schedule(protocol string, futureTimestamp int64, duration string, command string, args []string) {
+	//If future time is in past,run immediately
+	klog.Info("About to wait for futuretime:", futureTimestamp)
+	klog.Info("Current time:", time.Now().Unix())
+	time.Sleep(time.Duration(futureTimestamp-time.Now().Unix()) * time.Second)
+	startedAt = time.Now().Unix()
+	futureTime = futureTimestamp
+	tstDuration = duration
+	tstProtocol = protocol
+	execCmd(duration, command, args)
+}
+
+func execCmd(duration string, command string, args []string) {
+	cmd := exec.Command(command, args...)
+	out, err := cmd.StdoutPipe()
+	if err != nil {
+		klog.Error("unable to obtain Stdout:", err)
+		resultStatus <- err.Error()
+		return
+	}
+	eout, err := cmd.StderrPipe()
+	if err != nil {
+		klog.Error("unable to obtain Stderr:", err)
+		resultStatus <- err.Error()
+		return
+	}
+	multiRdr := io.MultiReader(out, eout)
+	err = cmd.Start()
+	if err != nil {
+		resultStatus <- err.Error()
+	}
+	scanOutput(&multiRdr)
+}
+
+func scanOutput(out *io.Reader) {
+	scanner := bufio.NewScanner(*out)
+	klog.Info("Starting scan for output")
+	for scanner.Scan() {
+		l := scanner.Text()
+		// klog.Info(line)
+		if l == "" {
+			continue
+		}
+		result = append(result, l)
+	}
+	klog.Info("Command executed,sending result back")
+	if err := scanner.Err(); err != nil {
+		klog.Error("Error", err)
+		resultStatus <- err.Error()
+		return
+	}
+	resultStatus <- "OK"
+}
+
+func parseResult() ([]float64, error) {
+	klog.Info("Parsing", result[0])
+	switch tstProtocol {
+	case protocolTCP:
+		return parseTCP(result), nil
+	case protocolUDP:
+		return parseUDP(result), nil
+	case protocolHTTP:
+		return parseHTTP(result), nil
+	default:
+		return nil, errors.New("invalid protocol:" + result[0])
+
+	}
+}
+
+func parseTCP(result []string) []float64 {
```

Comment:
Those parse* functions - can you move them to util.go file or sth like that

Reply:
Modified as suggested.